### ISSUE TITLE
Clarification of the Matching Loss

### ISSUE THREAD

**Issue by GloryyrolG** [facebookresearch/banmo#77]
Hi @gengshan-y @jhugestar et al.,

I have some questions about the matching loss. What if $\mathbf{X^{\ast}}$ is mapped into the BG in 3D space (due to an inaccurate warping), will not the matching loss under this circumstance mislead the learning of $\hat{\mathbf{X^{\ast}}}$? Thx & best,

![image](https://github.com/facebookresearch/banmo/assets/33202229/b8bfef04-b64e-4626-bde3-d1a287c06ccf)

![image](https://github.com/facebookresearch/banmo/assets/33202229/4763dfec-ce81-42d9-85e3-07ecf0cd735a)

>  To minimize the loss, the embeddings of surface points that do not project to (x, y) will be pulled away from the pixel embedding of (x, y) in a contrastive way [13]. (ViSER)

This seems to assume the projection is good enough?

**Comment by GloryyrolG** [comment]
cf https://github.com/facebookresearch/banmo/commit/0d507fe133443dca9bf078236a1f29748fbd73d4

**Comment by gengshan-y** [comment]
Hi, just to add on that

In viser, the 3D matching loss is used to overcome the trivial solution in the feature space, where "all pixel and surface features are the same constant, yielding zero reconstruction error". In banmo, we use the pre-trained densepose-cse as feature extractor, in which case feature-consistency loss does not lead to the trivial solution.
<img width="924" alt="image" src="https://github.com/facebookresearch/banmo/assets/13134872/d19ca345-8cad-401e-a89f-9db13d7e6b48">

**Comment by GloryyrolG** [comment]
Yeah, I understood. so it seems projection tsfm & correspondence are learned together. its inevitable that tsfm is not perfect at the beginning; which will be adjusted w/ correct correspondences learned at a later stage.
